vcd/resource_vcd_nsxt_edgegateway_bgp_configuration_test.go

```diff
@@ -0,0 +1,615 @@
+//go:build gateway || network || nsxt || ALL || functional
+// +build gateway network nsxt ALL functional
+
+package vcd
+
+import (
+	"fmt"
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+// TestAccVcdNsxtEdgeBgpConfigTier0 tests out NSX-T Edge Gateway BGP Configuration using dedicated
+// Tier-0 gateway
+func TestAccVcdNsxtEdgeBgpConfigTier0(t *testing.T) {
+	preTestChecks(t)
+	if !usingSysAdmin() {
+		t.Skip(t.Name() + " requires system admin privileges")
+		return
+	}
+
+	// Ensure Edge Gateway has a dedicated Tier 0 gateway (External network) as BGP and Route
+	// Advertisement configuration requires it. Restore it right after the test so that other
+	// tests are not impacted.
+	updateEdgeGatewayTier0Dedication(t, true)
+	defer updateEdgeGatewayTier0Dedication(t, false)
+
+	// String map to fill the template
+	var params = StringMap{
+		"Org":        testConfig.VCD.Org,
+		"NsxtVdc":    testConfig.Nsxt.Vdc,
+		"NsxtEdgeGw": testConfig.Nsxt.EdgeGateway,
+		"Tags":       "network nsxt",
+	}
+	testParamsNotEmpty(t, params)
+
+	// First step of test is going to alter some settings but not enable BGP because changing some of the fields
+	configText1 := templateFill(testAccVcdNsxtBgpConfig1, params)
+	debugPrintf("#[DEBUG] CONFIGURATION for step 1: %s", configText1)
+
+	params["FuncName"] = t.Name() + "-step2"
+	configText2 := templateFill(testAccVcdNsxtBgpConfig2, params)
+	debugPrintf("#[DEBUG] CONFIGURATION for step 2: %s", configText2)
+
+	params["FuncName"] = t.Name() + "-step3"
+	configText3 := templateFill(testAccVcdNsxtBgpConfig3DS, params)
+	debugPrintf("#[DEBUG] CONFIGURATION for step 3: %s", configText3)
+
+	params["FuncName"] = t.Name() + "-step4"
+	configText4 := templateFill(testAccVcdNsxtBgpConfig4, params)
+	debugPrintf("#[DEBUG] CONFIGURATION for step 4: %s", configText4)
+
+	params["FuncName"] = t.Name() + "-step5"
+	configText5 := templateFill(testAccVcdNsxtBgpConfig5, params)
+	debugPrintf("#[DEBUG] CONFIGURATION for step 5: %s", configText5)
+
+	params["FuncName"] = t.Name() + "-step6"
+	configText6 := templateFill(testAccVcdNsxtBgpConfig6DS, params)
+	debugPrintf("#[DEBUG] CONFIGURATION for step 6: %s", configText6)
+
+	params["FuncName"] = t.Name() + "-step8"
+	configText8 := templateFill(testAccVcdNsxtBgpConfig8, params)
+	debugPrintf("#[DEBUG] CONFIGURATION for step 8: %s", configText8)
+
+	params["FuncName"] = t.Name() + "-step9"
+	configText9 := templateFill(testAccVcdNsxtBgpConfig9, params)
+	debugPrintf("#[DEBUG] CONFIGURATION for step 9: %s", configText9)
+
+	if vcdShortTest {
+		t.Skip(acceptanceTestsSkipped)
+		return
+	}
+
+	resource.Test(t, resource.TestCase{
+		ProviderFactories: testAccProviders,
+		CheckDestroy:      testAccCheckNsxtBgpConfigurationDisabled(testConfig.Nsxt.EdgeGateway),
+		Steps: []resource.TestStep{
+			{
+				Config: configText1,
```

Comment:
could you add comments on what is tested in steps or changed between them

Reply:
I tried to add a bit more details.